test/cpp/util/byte_buffer_test.cc

```diff
@@ -27,6 +28,9 @@
 #include <gtest/gtest.h>
 
 namespace grpc {
+
+static internal::GrpcLibraryInitializer g_gli_initializer;
```

Comment:
Was there a reason to move all of the implementation of slice and bytebuffer into the headers? I assume that that is why there is an explicit initializer in the test. It looks a little awkward to have a library initializer in a test like this

Reply:
My reason for moving those implementations to headers was to make it reachable from codegen. There is a test, codegen_test_minimal, that is designed to ensure that cogegen doesn't pull in dependancies on grpc++. I initially had linker errors on that test, and inlining some of the slice and byte_buffer implementation seemed to be the best and cleanest fix.

The c implementations can hide behind core_codegen_interface, but we have no such mechanism for c++

Reply:
But I guess my question was what changed in codegen. The classes that you made public already existed in codegen as internal:: , so was it that the functions used by those classes changed, or something else?

Reply:
Ahh I understand the question now. I changed `GrpcProtoBufferReader` to take `ByteBuffer` instead of `grpc_byte_buffer`, so I needed to start using `ByteBuffer::Valid()` and `ByteBuffer::Length()` and the ctor.

And then in `GenericSerialize` I used `Slice()` ctor so I could use the `ByteBuffer tmp(Slice*, size)` ctor.